openeo-geotrellis/src/test/scala/org/openeo/geotrellis/layers/FileLayerProviderTest.scala

```diff
@@ -362,6 +371,16 @@ class FileLayerProviderTest {
     override def getCollections(correlationId: String): Seq[OpenSearchResponses.Feature] = ???
   }
 
+  class MockOpenSearchFeatures(mockedFeatures:Array[OpenSearchResponses.Feature]) extends OpenSearchClient {
+    override def getProducts(collectionId: String, dateRange: Option[(ZonedDateTime, ZonedDateTime)], bbox: ProjectedExtent, attributeValues: collection.Map[String, Any], correlationId: String, processingLevel: String): Seq[OpenSearchResponses.Feature] = {
+      mockedFeatures
+    }
+
+    override protected def getProductsFromPage(collectionId: String, dateRange: Option[(ZonedDateTime, ZonedDateTime)], bbox: ProjectedExtent, attributeValues: collection.Map[String, Any], correlationId: String, processingLevel: String, startIndex: Int): OpenSearchResponses.FeatureCollection = ???
+
+    override def getCollections(correlationId: String): Seq[OpenSearchResponses.Feature] = ???
+  }
```

Comment:
You can write the existing `MockOpenSearch` in terms of this `MockOpenSearchFeatures` to avoid code duplication.

Reply:
done